packages/amplify-provider-awscloudformation/src/utils/admin-login-server.ts

```diff
@@ -0,0 +1,127 @@
+import { stateManager } from 'amplify-cli-core';
+import { CognitoIdentity } from 'aws-sdk';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import express from 'express';
+import { JWK, JWKS, JWT } from 'jose';
+import _ from 'lodash';
+import fetch from 'node-fetch';
+
+import { TokenPayload, CognitoIdToken, CognitoAccessToken } from './cognitoJwtTypes';
+
+export class AdminLoginServer {
+  private port = 4242; // placeholder
+  private app: express.Application;
+  private server;
+  private appId: string;
+
+  private corsOptions: {
+    origin: string[];
+    methods: string[];
+    allowedHeaders: string;
+  };
+
+  constructor(appId: string, originUrl: string) {
+    this.appId = appId;
+    this.corsOptions = {
+      origin: [originUrl],
+      methods: ['POST', 'OPTIONS'],
+      allowedHeaders: 'Content-Type',
+    };
+    this.app = express();
+    this.app.use(cors(this.corsOptions));
+    this.app.use(bodyParser.json());
+  }
+
+  public async startServer(callback: () => void) {
+    await this.setupRoute(callback);
+    this.server = this.app.listen(this.getPort());
+  }
+
+  // TODO: scan for available ports across a range like mock
+  private getPort() {
+    return this.port;
+  }
+
+  private async getIdentityId(idToken: CognitoIdToken, IdentityPoolId: string, region: string): Promise<string> {
+    const cognitoIdentity = new CognitoIdentity({ region });
+    const login = idToken.payload.iss.replace('https://', '');
+    const logins = {
+      [login]: idToken.jwtToken,
+    };
+    const { IdentityId } = await cognitoIdentity
+      .getId({
+        IdentityPoolId,
+        Logins: logins,
+      })
+      .promise();
+    if (!IdentityId) {
+      throw new Error('IdentityId not defined. The CLI was unable to retrieve credentials.');
+    }
+    return IdentityId;
+  }
+
+  private async setupRoute(callback) {
+    this.app.post('/amplifyadmin/', async (req, res) => {
+      if (!req.body || req.body.error) {
+        this.shutdown();
+        if (req.body.error === 'CANCELLED') {
+          console.log('Login cancelled');
```

Comment:
Where are we printing these? If to the CLI's console output, then pass in context and print as usual.